pkg/services/login/login.go

```diff
@@ -90,6 +90,13 @@ func (ls *LoginService) UpsertUser(cmd *m.UpsertUserCommand) error {
 				return err
 			}
 		}
+
+		if extUser.AuthModule == models.AuthModuleLDAP && userQuery.Result.IsDisabled {
```

Comment:
I wonder if we have to actually check `auth` type? 

/cc @xlson

Reply:
I'm not sure this is strictly required, but I want to be explicit and mention that we should only enable/disable LDAP users (not OAuth or any other external provider).

Reply:
Scenario I'm trying to cover is:
1. User with login `ldap-user` logging into grafana through the LDAP auth.
2. User deleted from LDAP (and disabled in grafana)
3. User tries to login with OAuth (with the same `ldap-user` login)

I should take a look at `GetUserByAuthInfo()` function (will it return user in this case or not).

Reply:
I need to read the code in more depth to be completely certain, but I think we should be able to go for something like this:

```go
		if !extUser.IsDisabled && userQuery.Result.IsDisabled {
			
		}
```

If the user is disabled in the db but the external source claims that it isn't, we would re-enable it. I don't think we need to care about oauth and ldap co-existing with the same user id:s, that is a bad configuration and not one we need to support.

In this case, if the server admin wants to disable a user, they will have to do it by removing the user from oauth (or whatever they use).

What do you think?

Reply:
External source never returns `extUser.IsDisabled == true` because it's internal grafana flag.

Reply:
Yes. It would be the responsibility of each service to return the flag. Before we implement this each service would return `IsDisabled == false` as that is the default value.

Reply:
When auth has been moved out of Grafana (as with LDAP and OAuth) it doesn't really make sense to be able to control the disabled status of a user within Grafana.